File: demo/src/main/java/com/example/demo/dto/AnnonceResponseDTO.java
package com.example.demo.dto;

import com.example.demo.enums.Categorie;
import com.example.demo.enums.Disponibilite;
import com.example.demo.enums.StatutLivraison;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class AnnonceResponseDTO {
    private Long id;
    private String title;
    private String description;
    private double price;
    private Categorie category;
    private Disponibilite disponibilite;
    private LocalDateTime creationDate;
    private Long vendeurId;
    private String vendeurName;
    private String vendeurEmail;
    private Long acheteurId;
    private List<String> images;
    private Long livraisonId;
    private StatutLivraison statutLivraison;
}

File: OccaPlace_Angular/src/app/core/dto/AnnonceResponseDTO.ts
import {Categorie} from "../enum/Categorie";
import {Disponibilite} from "../enum/Disponibilite";
import {StatutLivraison} from "../enum/StatutLivraison";


export interface AnnonceResponseDTO {
  id: number;
  title: string;
  description: string;
  price: number;
  category: Categorie;
  disponibilite: Disponibilite;
  creationDate: string;
  vendeurId: number;
  vendeurName?: string;
  vendeurEmail: string;
  acheteurId?: number;
  livraisonId?: number;
  statutLivraison?: StatutLivraison;
  images: string[];
}
